config/proxy/proxy.go

```diff
@@ -0,0 +1,197 @@
+// Description: This file contains the implementation of the new consolidated proxy/geolocation config.
+
+package proxyconfig
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/getlantern/eventual/v2"
+	"github.com/getlantern/golog"
+	"github.com/getlantern/rot13"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/getlantern/flashlight/v7/apipb"
+)
+
+const (
+	defaultConfigSaveDir  = ""
+	defaultConfigFilename = "proxies.conf"
+)
+
+// alias for better readability
+type ProxyConfig = apipb.ConfigResponse
+
+var (
+	_config = &config{
+		config: eventual.NewValue(),
+	}
+
+	log = golog.LoggerFor("proxyconfig")
+)
+
+type config struct {
+	config eventual.Value
+
+	// filePath is where we should save new configs and look for existing saved configs.
+	filePath string
+	// obfuscate specifies whether or not to obfuscate the config on disk.
+	obfuscate bool
+
+	// listeners is a list of functions to call when the config changes.
+	listeners []func(old, new *ProxyConfig)
+	mu        sync.Mutex
+}
+
+func Init(saveDir string, obfuscate bool) (*config, error) {
+	if saveDir == "" {
+		saveDir = defaultConfigSaveDir
+		_config.filePath = filepath.Join(saveDir, defaultConfigFilename)
+	}
+
+	_config.obfuscate = obfuscate
+
+	saved, err := readExistingConfig(_config.filePath, obfuscate)
+	if err != nil {
+		log.Errorf("Failed to read existing client config: %w", err)
+	}
+
+	_config.config.Set(saved)
+	notifyListeners(nil, saved)
+	return _config, nil
+}
+
+// GetConfig implements services.ConfigHandler
+func (c *config) GetConfig() *ProxyConfig {
+	conf, _ := c.config.Get(eventual.DontWait)
+	return conf.(*ProxyConfig)
+}
+
+// SetConfig implements services.ConfigHandler
+func (c *config) SetConfig(new *ProxyConfig) {
+	old := c.GetConfig()
+	c.config.Set(new)
+	if err := saveConfig(c.filePath, new, c.obfuscate); err != nil {
+		log.Errorf("Failed to save client config: %w", err)
+	}
+
+	notifyListeners(old, new)
+}
+
+// GetConfig returns the current client config. An error is returned if the config is not available
+// within the given timeout.
+func GetConfig(timeout time.Duration) (*ProxyConfig, error) {
```

Comment:
I thought about that, but I wanted to hide the use of `eventual` from callers as they don't need to know internals. I have no problem changing it though if you think it would be better to accept a context instead.